docs/cookbook/using-docker.md

```diff
@@ -0,0 +1,340 @@
+
+### Creating the install script
+
+The alpine linux distribution is slim and doesn't include bash, but we can access a the "busybox" shell, which is compatible with the basics, at `/bin/sh`. Into the `scripts` folder at the root of your project create the following file:
+
+<AposCodeBlock>
+
+```bash
+#!/bin/sh
+
+export APOS_RELEASE_ID=`cat /dev/urandom |env LC_CTYPE=C tr -dc 'a-zA-Z0-9' | fold -w 32 | head -n 1`
+
+echo $APOS_RELEASE_ID > ./release-id
+
+node app @apostrophecms/asset:build
+```
+
+<template v-slot:caption>
+  scripts/build-asset.sh
+</template>
+
+</AposCodeBlock>
+
+We won't go through this file in detail. As covered in the previous section, it creates a random unique string and copies it out to the `release-id` file at the root of the project. It then triggers the `@apostrophecms/asset` module to build the assets. That module will read the `release-id` file and use the string in the build.
+
+Building the assets inside this script, which is part of a **build step** in the Dockerfile, ensures the assets become part of the image, so they don't have to be re-generated every time the image is used. The same is true for the `release-id` file, which Apostrophe uses to identify the asset bundle it should be using. It'll be the same bundle at build time and at run time. If the image is rebuilt, we'll get a new image, new CSS URLs, and no stale stylesheets.
+
+### Creating a `.dockerignore` file
+The `.dockerignore` file prevents specific files from being copied into your final image. This is important to block sensitive or unnecessary files from being incorporated into your image. Simply go through your directory and copy any file or folder name not needed to build your project into your `.dockerignore` file. Note that folder names are followed by a `/`. I'm using Visual Studio Code in this tutorial, so the topmost folder listed won't be in your project if you use a different editor.
+
+<AposCodeBlock>
+
+```bash
+.vscode/
+apos-build/
+badges/
+data/
+node-modules/
+public/uploads/
+.dockerignore
+.env
+.eslintignore
+.gitignore
+deploy-test-count
+docker-compose.yaml
+dockerfile
+force-deploy
+local.example.js
+```
+
+<template v-slot:caption>
+  .dockerignore
+</template>
+
+</AposCodeBlock>
+
+### Creating a `docker-compose.yaml` file
+In this guide, we are starting by creating multiple containers and a persistent volume. This is so that we can provide both a MongoDB instance and a place to store uploaded assets. We are going to do this using [Docker Compose](https://docs.docker.com/compose/) and a `docker-compose.yml` file. In the following sections of the tutorial, we will look at removing the extra container and volume by taking advantage of cloud storage and database services. Create this file at the root of your project.
+
+<AposCodeBlock>
+
+```bash
+services:
+  db:
+    image: mongo:4.4.14
+    ports:
+      - "27017:27017"
```

Comment:
Let's just go with 27018 everywhere, and set APOS_MONGODB_URI to `mongodb://localhost:27018/project` inside the container

Reply:
Changed.